tools/rtpsrelay/lib/QosIndex.h

```diff
@@ -184,20 +310,28 @@ class Literal {
     index_.erase(writer);
   }
 
-  void insert_literal(ReaderPtr reader)
+  void insert_literal(ReaderPtr reader,
+                      GuidSet& local_guids,
+                      RelayAddressesSet& relay_addresses)
   {
-    index_.insert(reader);
     ++literal_count_;
+    index_.insert(reader, local_guids, relay_addresses);
   }
 
-  void insert_pattern(ReaderPtr reader)
+  void insert_pattern(ReaderPtr reader,
+                      GuidSet& local_guids,
+                      RelayAddressesSet& relay_addresses)
   {
-    index_.insert(reader);
+    return index_.insert(reader, local_guids, relay_addresses);
```

Comment:
```suggestion
    index_.insert(reader, local_guids, relay_addresses);
```

Reply:
You are better than a compiler.

Reply:
In some ways yes, in others, no.  My debuginfo is OK but inter-procedural optimization not so much.